src/zkevm_specs/state.py

```diff
@@ -127,21 +121,17 @@ def check_memory(row: Row, row_prev: Row):
     if not all_keys_eq(row, row_prev) and row.is_write == 0:
         assert row.value == 0
 
-    # 2. mem_addr in range
-    mem_addr = get_mem_addr(row)
-    assert_in_range(mem_addr, 0, MAX_MEMORY_ADDRESS)
-
-    # 3. value is a byte
+    # 2. value is a byte
```

Comment:
Following my comment about defining memory address as 32 bits.

I'm not sure if the EVM circuit will make sure that the memory address is range constrained already, but we could easily add a range constraint here by constraining that `key2_limbs[2:]` are 0.

Reply:
The only parts of the evm circuit that constraint the memory address size are related to the gas cost calculations. Now that I think about it, it's possible that the gas cost calculation could overflow for memory addresses > 2^127 because of the square in the formula.

Reply:
we won't have to worry about this with the 32 bit constraint for memory addresses.